src/charm.py

```diff
@@ -71,10 +71,8 @@ def __init__(self, *args):
             redis_relation={},
         )
         self.ingress = IngressRequires(self, self._make_ingress_config())
-        self.framework.observe(self.on.leader_elected, self._config_changed)
         self.framework.observe(self.on.discourse_pebble_ready, self._config_changed)
         self.framework.observe(self.on.config_changed, self._config_changed)
-        self.framework.observe(self.on.upgrade_charm, self._config_changed)
```

Comment:
I think this is needed as when the charm gets updated, the application could be updated too and the pod_setup needs to run in order to proceed with the DB migrations